### PR TITLE
[Aptos Framework][Prover] Made the Aptos Framework compatible with Prover

### PR DESCRIPTION
**Pull request by junkil-park**
This commit makes the Aptos Framework compatible with Prover.
  - fixed the existing spec that has been broken
  - mocked up the native functions and the xor operation so that Prover does not crash.

`aptos move prove` works without any error/crash for all of the following Move packages:
  - `aptos-framework`
  - `aptos-token`
  - `aptos-stdlib`
  - `move-stdlib`

TODO:
- add a new spec to the Stake module
- add the boogie implementation for the native functions which are temporarily mocked up in this PR.

### Test Plan
Run `aptos move prove` in any of the Aptos Move packages above.

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2162)

### CODE REVIEW

**Review by davidiw — COMMENTED** [review]
reads pretty straight forward, will see if @wrwg has any feedback since this is a bit foreign to me.

**Review by wrwg — APPROVED** [review]
Great!

**Comment by wrwg** [comment]
You need to add the `CICD:run-e2e-tests` label, then do a force push (I believe) to let the forge tests run. The PR otherwise waits for those tests to complete, but they are not run without the label.

File: aptos-move/framework/aptos-framework/sources/timestamp.move [hunk 1]
```diff
@@ -46,17 +45,14 @@ module aptos_framework::timestamp {
         /// and need to hold.
         pragma delegate_invariants_to_caller;
         include AbortsIfNotGenesis;
-        include system_addresses::AbortsIfNotaptos_framework{account: signer::address_of(account)};
```
Comment (movekevin) [inline aptos-move/framework/aptos-framework/sources/timestamp.move:49]: Nice! Thanks for the cleanup :)

File: aptos-move/framework/aptos-stdlib/Move.toml [hunk 2]
```diff
@@ -5,6 +5,7 @@ version = "1.0.0"
 [addresses]
 std = "0x1"
 aptos_std = "0x1"
+Extensions = "0x1" # TODO: Needed for Prover to instantiate `{{Ext}}` in prelude.
```
Comment (movekevin) [inline aptos-move/framework/aptos-stdlib/Move.toml:8]: Why is this a TODO? It sounds like the prover just needs it.

**Review by movekevin — APPROVED** [review]

**Comment by movekevin** [comment]
> You need to add the `CICD:run-e2e-tests` label, then do a force push (I believe) to let the forge tests run. The PR otherwise waits for those tests to complete, but they are not run without the label.

Alternatively, you can enable auto-merge since the PR is ready and approved anyway. This will also trigger forge.

File: aptos-move/framework/aptos-stdlib/Move.toml (hunk 2 above)
Comment (junkil-park) [reply aptos-move/framework/aptos-stdlib/Move.toml:8]: I will see if there is a better way to handle this in Prover.

**Pull request merged**